src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -4240,7 +4284,7 @@ && serverFeaturesModified()) {
         } else if (isDockerfileDirectoryChanged(fileChanged)) {
             // If contents within a directory specified in a Dockerfile COPY command were changed, and not already processed by one of the other conditions above.
             untrackDockerfileDirectoriesAndRestart();
-        } else if (generateFeatures && directory.startsWith(srcOutputPath)) {
+        } else if (!recompileDependencies && generateFeatures && directory.startsWith(srcOutputPath)) {
```

Comment:
Suggestion to add some comments indicating why we track class files when `recompileDependencies=false` and what the new behaviour is when `recompileDependencies=true`